src/rules/html-self-closing.ts

```diff
@@ -0,0 +1,150 @@
+import type { AST } from "svelte-eslint-parser"
+import { createRule } from "../utils"
+import { getNodeName, isVoidHtmlElement } from "../utils/template-utils"
+
+const TYPE_MESSAGES = {
+  normal: "HTML elements",
+  void: "HTML void elements",
+  component: "Svelte custom components",
+  svelte: "Svelte special elements",
+}
+
+type ElementTypes = "normal" | "void" | "component" | "svelte"
+
+export default createRule("html-self-closing", {
+  meta: {
+    docs: {
+      description: "enforce self-closing style",
+      category: "Stylistic Issues",
+      recommended: false,
+      conflictWithPrettier: true,
+    },
+    type: "layout",
+    fixable: "code",
+    messages: {
+      requireClosing: "Require self-closing on {{type}}.",
+      disallowClosing: "Disallow self-closing on {{type}}.",
+    },
+    schema: [
+      {
+        type: "object",
+        properties: {
+          html: {
+            type: "object",
+            properties: {
+              void: {
+                enum: ["never", "always", "ignore"],
+              },
+              normal: {
+                enum: ["never", "always", "ignore"],
+              },
+              component: {
+                enum: ["never", "always", "ignore"],
+              },
+              svelte: {
+                enum: ["never", "always", "ignore"],
+              },
+            },
+            additionalProperties: false,
+          },
+        },
+        additionalProperties: false,
+      },
+    ],
+  },
+  create(ctx) {
+    const options = {
+      ...ctx.options?.[0],
+      html: {
+        void: "always",
+        normal: "always",
+        component: "always",
+        svelte: "always",
+        ...ctx.options?.[0]?.html,
+      },
```

Comment:
Currently we only have the `html` option, so we don't need the rest element.

```suggestion
      html: {
        void: "always",
        normal: "always",
        component: "always",
        svelte: "always",
        ...ctx.options?.[0]?.html,
      },
```